### DISCUSSION TITLE
Java 17 Text Blocks - CONSTRUCT query - Invalid characters (CR/LF) in header query

### DISCUSSION THREAD

**Discussion by naturzukunft** [eclipse-rdf4j/rdf4j#3653] [Q&A] [answered]
Hi there, currently playing with "Java Text Blocks" for CONSTRUCT Query. 
i got an error, because the query is sent via header attribute:

`Model m = Repositories.graphQuery(rep, query, r -> QueryResults.asModel(r));`

so the query is long, but not long enough to be sent via post. Is there a way to force http post ?

**Comment by abrokenjester** [comment, marked as answer]
The query itself is never sent via a header attribute. It's either sent as a url parameter (when using a GET request), or as part of the payload body (when using POST). And as you correctly remark, the choice depends on the total length of the URL (including the parameters): if the client detects that that goes over a limit, it switches to POST. By default, that value is set to 4083 characters. 

If your HTTP server is not accepthing URLs of that length though, you can reconfigure the max length default by setting the `rdf4j.sparql.url.maxlength` environment variable to some lower value. You can even set it to 0, in which case it will use POST for _every_ query. See also https://rdf4j.org/javadoc/latest/org/eclipse/rdf4j/http/client/SPARQLProtocolSession.html#MAXIMUM_URL_LENGTH_PARAM

**Reply by naturzukunft** [reply]
this is exactly what I was looking for
